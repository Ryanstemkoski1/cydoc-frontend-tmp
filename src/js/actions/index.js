import {ADD_NOTE, ADD_TEMPLATE, DATA_LOADED, LOGIN_REQUEST, LOGOUT, SAVE_NOTE} from "../constants/action-types";
import axios from 'axios'
import api from "../constants/api";

export function addNote(payload) {
    return { type: ADD_NOTE, payload };
}

export function addTemplate(payload) {
    return { type: ADD_TEMPLATE, payload: payload}
}

export function getAllRecords() {
    return function(dispatch) {
        return axios.get(api.records.dev)
            .then(response => response.data)
            .then(data => {
                dispatch({ type: DATA_LOADED, payload: data });
            });
    };
}

export function loginRequest(payload){
    return function(dispatch) {
        return axios.post(api.login.dev, payload)
            .then(res => {
                const user = res;
                console.log(user)
                localStorage.setItem('user', JSON.stringify(user));
                return user;
            })
            .then((user)=> {
                dispatch({type: LOGIN_REQUEST, payload: user.data})
                return user;
            })
            .catch(err => {
                return err.response;
            })
    }
}

export function logout(){
        localStorage.removeItem('user');
        return { type: LOGOUT };
}

export  function saveNote(payload){
    return function (dispatch) {
        const headers = {
            'Content-Type': 'application/json',

        };
        return axios.post(api.newRecord.dev,
            { headers: {
                'Content-Type': 'application/json'}},
            payload)
            .then(res => res.data)
            .then(data => {
                dispatch({type: SAVE_NOTE, payload: data})
            })
    }
}