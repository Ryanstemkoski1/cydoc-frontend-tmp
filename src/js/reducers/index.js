// src/js/reducers/index.js
import {ADD_NOTE, DATA_LOADED, LOGIN_REQUEST, LOGOUT} from "../constants/action-types";

const initialState = {
    notes: [],
    remoteRecords: [],
    user: {}
};

let user = JSON.parse(localStorage.getItem('user'));
const initialUserState = user ? { loggedIn: true, user } : {};
initialState[user] = initialUserState

function rootReducer(state = initialState, action) {
    if (action.type === ADD_NOTE) {
        return Object.assign({}, state, {
            notes: state.notes.concat(action.payload)
        });
    }
    if (action.type === DATA_LOADED) {
        return Object.assign({}, state, {
            remoteRecords: state.remoteRecords.concat(action.payload)
        });
    }

    if(action.type === LOGIN_REQUEST) {
        return Object.assign({}, state, {
            user: state.user.concat(action.payload)
        })
    }

    if(action.type === LOGOUT) {
        return {};
    }

    return state;
}
export default rootReducer;