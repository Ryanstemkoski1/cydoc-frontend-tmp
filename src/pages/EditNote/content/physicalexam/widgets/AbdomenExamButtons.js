import React, {Component} from 'react'
import { Button, Popup } from 'semantic-ui-react'
import HPIContext from 'contexts/HPIContext.js'

export default class AbdomenExamButtons extends Component {
    static contextType = HPIContext 
    constructor(props, context) {
        super(props, context)
        this.onClick = this.onClick.bind(this)
    }

    onClick(event, data) {
        var values = this.context["Physical Exam"]
        var prevState = values.widgets["Abdomen"][this.props.ab_quadrant][data.children]
        values.widgets["Abdomen"][this.props.ab_quadrant][data.children] = !prevState
        this.context.onContextChange("Physical Exam", values);
    }

    render() {
        var ab_quadrant_buttons = []
        var ab_quadrant_dict = this.context["Physical Exam"].widgets["Abdomen"][this.props.ab_quadrant]
        for (var ab_quadrant_button in ab_quadrant_dict) {
            ab_quadrant_buttons.push(
                <Button fluid
                    size = {'small'}
                    key={this.props.ab_quadrant + " " + ab_quadrant_button}
                    color={this.context["Physical Exam"].widgets["Abdomen"][this.props.ab_quadrant][ab_quadrant_button] ? 'red' : ''}
                    onClick={this.onClick}
                    style = {{marginBottom: 5}}
                > 
                    {ab_quadrant_button} 
                </Button>
            )
        } 
        return (
                <div>{ab_quadrant_buttons}</div>
              )
        }
    }

