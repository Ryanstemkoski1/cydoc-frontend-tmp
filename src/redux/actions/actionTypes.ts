export enum CURRENT_NOTE_ACTION {
    LOAD_NOTE = 'CURRENT_NOTE_LOAD_NOTE',
    DELETE_NOTE = 'CURRENT_NOTE_DELETE_NOTE',
    UPDATE_NOTE_TITLE = 'CURRENT_NOTE_UPDATE_TITLE',
    UPDATE_NOTE_ID = 'CURRENT_NOTE_UPDATE_ID',
}

export enum ROS_ACTION {
    TOGGLE_OPTION = 'ROS_TOGGLE_OPTION',
}

export enum PHYSICAL_EXAM_ACTION {
    UPDATE_VITALS = 'PHYSICAL_EXAM_UPDATE_VITALS',
    TOGGLE_FINDING = 'PHYSICAL_EXAM_TOGGLE_FINDING',
    TOGGLE_LEFT_RIGHT_FINDING = 'PHYSICAL_EXAM_TOGGLE_LEFT_RIGHT_FINDING',
    TOGGLE_CHOOSE_BOOLEAN_VALUE = 'PHYSICAL_EXAM_TOGGLE_CHOOSE_BOOLEAN_VALUE',
    REMOVE_FINDING = 'PHYSICAL_EXAM_REMOVE_FINDING',
    UPDATE_COMMENTS = 'PHYSICAL_EXAM_UPDATE_COMMENTS',
}

export enum LUNGS_WIDGET_ACTION {
    TOGGLE_LUNGS_WIDGET_SECTION = 'LUNGS_WIDGET_TOGGLE_LUNGS_WIDGET_SECTION',
}

export enum ABDOMEN_WIDGET_ACTION {
    TOGGLE_ABDOMEN_WIDGET_SECTION = 'ABDOMEN_WIDGET_TOGGLE_ABDOMEN_WIDGET_SECTION',
}

export enum PULSES_WIDGET_ACTION {
    ADD_PULSES_WIDGET_ITEM = 'PULSES_WIDGET_ADD_PULSES_WIDGET_ITEM',
    UPDATE_LOCATION = 'PULSES_WIDGET_UPDATE_LOCATION',
    UPDATE_SIDE = 'PULSES_WIDGET_UPDATE_SIDE',
    UPDATE_INTENSITY = 'PULSES_WIDGET_UPDATE_INTENSITY',
    DELETE_PULSES_WIDGET_ITEM = 'PULSES_WIDGET_DELETE_PULSES_WIDGET_ITEM',
}

export enum REFLEXES_WIDGET_ACTION {
    ADD_REFLEXES_WIDGET_ITEM = 'REFLEXES_WIDGET_ADD_REFLEXES_WIDGET_ITEM',
    UPDATE_LOCATION = 'REFLEXES_WIDGET_UPDATE_LOCATION',
    UPDATE_SIDE = 'REFLEXES_WIDGET_UPDATE_SIDE',
    UPDATE_INTENSITY = 'REFLEXES_WIDGET_UPDATE_INTENSITY',
    DELETE_REFLEXES_WIDGET_ITEM = 'REFLEXES_WIDGET_DELETE_REFLEXES_WIDGET_ITEM',
}

export enum MURMURS_WIDGET_ACTION {
    ADD_MURMURS_WIDGET_ITEM = 'MURMURS_WIDGET_ADD_MURMURS_WIDGET_ITEM',
    UPDATE_PHASE = 'MURMURS_WIDGET_UPDATE_PHASE',
    TOGGLE_CRESCENDO_DECRESCENDO = 'MURMURS_WIDGET_TOGGLE_CRESCENDO_DECRESCENDO',
    UPDATE_BEST_HEARD_AT = 'MURMURS_WIDGET_UPDATE_BEST_HEARD_AT',
    UPDATE_INTENSITY = 'MURMURS_WIDGET_UPDATE_INTENSITY',
    UPDATE_PITCH = 'MURMURS_WIDGET_UPDATE_PITCH',
    TOGGLE_QUALITY = 'MURMURS_WIDGET_TOGGLE_QUALITY',
    TOGGLE_SPECIFIC_MURMUR_INFO = 'MURMURS_WIDGET_TOGGLE_SPECIFIC_MURMUR_INFO',
    UPDATE_SPECIFIC_MURMUR = 'MURMURS_WIDGET_UPDATE_SPECIFIC_MURMUR',
    TOGGLE_RADIATION_TO = 'MURMURS_WIDGET_TOGGLE_RADIATION_TO',
    TOGGLE_ADDITIONAL_FEATURES = 'MURMURS_WIDGET_TOGGLE_ADDITIONAL_FEATURES',
    DELETE_MURMURS_WIDGET_ITEM = 'MURMURS_WIDGET_DELETE_MURMURS_WIDGET_ITEM',
}
export enum MEDICAL_HISTORY_ACTION {
    TOGGLE_OPTION = 'MEDICAL_HISTORY_TOGGLE_OPTION',
    UPDATE_CONDITION_NAME = 'MEDICAL_HISTORY_UPDATE_CONDITION_NAME',
    UPDATE_START_YEAR = 'MEDICAL_HISTORY_UPDATE_START_YEAR',
    UPDATE_CONDITION_RESOLVED = 'MEDICAL_HISTORY_UPDATE_CONDITION_RESOLVED',
    UPDATE_END_YEAR = 'MEDICAL_HISTORY_UPDATE_END_YEAR',
    UPDATE_COMMENTS = 'MEDICAL_HISTORY_UPDATE_COMMENTS',
    ADD_PMH_POP_OPTIONS = 'MEDICAL_HISTORY_ADD_PMH_POP_OPTIONS',
    DELETE_CONDITION = 'MEDICAL_HISTORY_DELETE_CONDITION',
}

export enum SURGICAL_HISTORY_ACTION {
    UPDATE_PROCEDURE = 'SURGICAL_HISTORY_UPDATE_PROCEDURE',
    TOGGLE_OPTION = 'SURGICAL_HISTORY_TOGGLE_OPTION',
    UPDATE_YEAR = 'SURGICAL_HISTORY_UPDATE_YEAR',
    UPDATE_COMMENTS = 'SURGICAL_HISTORY_UPDATE_COMMENTS',
    DELETE_PROCEDURE = 'SURGICAL_HISTORY_DELETE_PROCEDURE',
    ADD_PSH_POP_OPTIONS = 'SURGICAL_HISTORY_ADD_PSH_POP_OPTIONS',
}

export enum MEDICATIONS_ACTION {
    UPDATE_DRUG_NAME = 'MEDICATIONS_UPDATE_DRUG_NAME',
    UPDATE_START_YEAR = 'MEDICATIONS_UPDATE_START_YEAR',
    UPDATE_CURRENTLY_TAKING = 'MEDICATIONS_UPDATE_CURRENTLY_TAKING',
    UPDATE_END_YEAR = 'MEDICATIONS_UPDATE_END_YEAR',
    UPDATE_SCHEDULE = 'MEDICATIONS_SCHEDULE',
    UPDATE_DOSE = 'MEDICATIONS_UPDATE_DOSE',
    UPDATE_REASON_FOR_TAKING = 'MEDICATIONS_UPDATE_REASON_FOR_TAKING',
    UPDATE_SIDE_EFFECTS = 'MEDICATIONS_UPDATE_SIDE_EFFECTS',
    UPDATE_COMMENTS = 'MEDICATIONS_UPDATE_COMMENTS',
    DELETE_MEDICATION = 'MEDICATIONS_DELETE_MEDICATION',
    ADD_MEDS_POP_OPTION = 'MEDICATIONS_ADD_MEDS_POP_OPTION',
}

export enum ALLERGIES_ACTION {
    UPDATE_INCITING_AGENT = 'ALLERGIES_UPDATE_INCITING_AGENT',
    UPDATE_REACTION = 'ALLERGIES_UPDATE_REACTION',
    UPDATE_COMMENTS = 'ALLERGIES_UPDATE_COMMENTS',
    UPDATE_ID = 'ALLERGIES_UPDATE_ID',
    ADD_ALLERGY = 'ALLERGIES_ADD_ALLERGY',
    DELETE_ALLERGY = 'ALLERGIES_DELETE_ALLERGY',
}

export enum SOCIAL_HISTORY_ACTION {
    UPDATE_ALCOHOL_USAGE = 'SOCIAL_HISTORY_UPDATE_A_USAGE',
    ADD_ALCOHOL_CONSUMPTION = 'SOCIAL_HISTORY_ADD_A_CONSUMPTION',
    UPDATE_ALCOHOL_CONSUMPTION_TYPE = 'SOCIAL_HISTORY_UPDATE_A_CONSUMPTION_TYPE',
    UPDATE_ALCOHOL_CONSUMPTION_SIZE = 'SOCIAL_HISTORY_UPDATE_A_CONSUMPTION_SIZE',
    UPDATE_ALCOHOL_CONSUMPTION_NUMBER_PER_WEEK = 'SOCIAL_HISTORY_UPDATE_A_CONSUMPTION_PER_WEEK',
    DELETE_ALCOHOL_CONSUMPTION = 'SOCIAL_HISTORY_DELETE_A_CONSUMPTION',
    UPDATE_ALCOHOL_QUIT_YEAR = 'SOCIAL_HISTORY_UPDATE_A_QUIT_YEAR',
    UPDATE_ALCOHOL_COMMENTS = 'SOCIAL_HISTORY_UPDATE_A_COMMENTS',
    UPDATE_ALCOHOL_INTERESTED_IN_QUITTING = 'SOCIAL_HISTORY_UPDATE_A_IIQ',
    UPDATE_ALCOHOL_TRIED_TO_QUIT = 'SOCIAL_HISTORY_UPDATE_A_TTQ',

    UPDATE_TOBACCO_USAGE = 'SOCIAL_HISTORY_UPDATE_T_USAGE',
    UPDATE_TOBACCO_PACKS_PER_DAY = 'SOCIAL_HISTORY_UPDATE_T_PACKS_PER_DAY',
    UPDATE_TOBACCO_NUMBER_OF_YEARS = 'SOCIAL_HISTORY_UPDATE_T_NUMBER_OF_YEARS',
    UPDATE_TOBACCO_PRODUCTS_USED = 'SOCIAL_HISTORY_UPDATE_T_PRODUCTS_USED',
    UPDATE_TOBACCO_QUIT_YEAR = 'SOCIAL_HISTORY_UPDATE_T_QUIT_YEAR',
    UPDATE_TOBACCO_COMMENTS = 'SOCIAL_HISTORY_UPDATE_T_COMMENTS',
    UPDATE_TOBACCO_INTERESTED_IN_QUITTING = 'SOCIAL_HISTORY_UPDATE_T_IIQ',
    UPDATE_TOBACCO_TRIED_TO_QUIT = 'SOCIAL_HISTORY_UPDATE_T_TTQ',

    UPDATE_RECREATIONAL_DRUG_USAGE = 'SOCIAL_HISTORY_UPDATE_RD_USAGE',
    ADD_RECREATIONAL_DRUG_USED = 'SOCIAL_HISTORY_ADD_RD_USED',
    UPDATE_RECREATIONAL_DRUG_USED_NAME = 'SOCIAL_HISTORY_UPDATE_RD_USED_NAME',
    UPDATE_RECREATIONAL_DRUG_USED_MODES_OF_DELIVERY = 'SOCIAL_HISTORY_UPDATE_RD_USED_MOD',
    UPDATE_RECREATIONAL_DRUG_USED_NUMBER_PER_WEEK = 'SOCIAL_HISTORY_UPDATE_RD_USED_PER_WEEK',
    DELETE_RECREATIONAL_DRUG_USED = 'SOCIAL_HISTORY_DELETE_RD_USED',
    UPDATE_RECREATIONAL_DRUG_QUIT_YEAR = 'SOCIAL_HISTORY_UPDATE_RD_QUIT_YEAR',
    UPDATE_RECREATIONAL_DRUG_COMMENTS = 'SOCIAL_HISTORY_UPDATE_RD_COMMENTS',
    UPDATE_RECREATIONAL_DRUG_INTERESTED_IN_QUITTING = 'SOCIAL_HISTORY_UPDATE_RD_IIQ',
    UPDATE_RECREATIONAL_DRUG_TRIED_TO_QUIT = 'SOCIAL_HISTORY_UPDATE_RD_TTQ',

    UPDATE_LIVING_SITUATION = 'SOCIAL_HISTORY_UPDATE_LIVING_SITUATION',
    UPDATE_EMPLOYMENT = 'SOCIAL_HISTORY_UPDATE_EMPLOYMENT',
    UPDATE_DIET = 'SOCIAL_HISTORY_UPDATE_DIET',
    UPDATE_EXERCISE = 'SOCIAL_HISTORY_UPDATE_EXERCISE',
}

export enum FAMILY_HISTORY_ACTION {
    TOGGLE_CONDITION_OPTION = 'FAMILY_HISTORY_TOGGLE_CONDITION_OPTION',
    ADD_FAMILY_MEMBER = 'FAMILY_HISTORY_ADD_FAMILY_MEMBER',
    DELETE_FAMILY_MEMBER = 'FAMILY_HISTORY_DELETE_FAMILY_MEMBER',
    UPDATE_MEMBER = 'FAMILY_HISTORY_UPDATE_MEMBER',
    TOGGLE_CAUSE_OF_DEATH_OPTION = 'FAMILY_HISTORY_TOGGLE_CAUSE_OF_DEATH_OPTION',
    TOGGLE_LIVING_OPTION = 'FAMILY_HISTORY_TOGGLE_LIVING_OPTION',
    UPDATE_COMMENTS = 'FAMILY_HISTORY_UPDATE_COMMENTS',
    UPDATE_CONDITION_NAME = 'FAMILY_HISTORY_UPDATE_CONDITION_NAME',
    ADD_FH_POP_OPTIONS = 'FAMILY_HISTORY_ADD_FH_POP_OPTIONS',
    DELETE_CONDITION = 'FAMILY_HISTORY_DELETE_CONDITION',
}

export enum PLAN_ACTION {
    ADD_CONDITION = 'PLAN_ADD_CONDITION',
    DELETE_CONDITION = 'PLAN_DELETE_CONDITION',
    UPDATE_CONDITION_NAME = 'PLAN_UPDATE_CONDITION_NAME',

    ADD_DIFFERENTIAL_DIAGNOSIS = 'PLAN_ADD_DIFFERENTIAL_DIAGNOSIS',
    UPDATE_DIFFERENTIAL_DIAGNOSIS_COMMENTS = 'PLAN_UPDATE_DIFFERENTIAL_DIAGNOSIS_COMMENTS',
    UPDATE_DIFFERENTIAL_DIAGNOSIS = 'PLAN_UPDATE_DIFFERENTIAL_DIAGNOSIS',
    DELETE_DIFFERENTIAL_DIAGNOSIS = 'PLAN_DELETE_DIFFERENTIAL_DIAGNOSIS',

    ADD_PRESCRIPTION = 'PLAN_ADD_PRESCRIPTION',
    UPDATE_PRESCRIPTION_COMMENTS = 'PLAN_UPDATE_PRESCRIPTION_COMMENTS',
    UPDATE_PRESCRIPTION_DOSE = 'PLAN_UPDATE_PRESCRIPTION_DOSE',
    UPDATE_PRESCRIPTION_TYPE = 'PLAN_UPDATE_PRESCRIPTION_TYPE',
    UPDATE_PRESCRIPTION_SIGNATURE = 'PLAN_UPDATE_PRESCRIPTION_SIGNATURE',
    DELETE_PRESCRIPTION = 'PLAN_DELETE_PRESCRIPTION',

    ADD_PROCEDURE_OR_SERVICE = 'PLAN_ADD_PROCEDURE_OR_SERVICE',
    UPDATE_PROCEDURE_OR_SERVICE = 'PLAN_UPDATE_PROCEDURE_OR_SERVICE',
    UPDATE_PROCEDURE_OR_SERVICE_COMMENTS = 'PLAN_UPDATE_PROCEDURE_OR_SERVICE_COMMENTS',
    UPDATE_PROCEDURE_OR_SERVICE_WHEN = 'PLAN_UPDATE_PROCEDURE_OR_SERVICE_WHEN',
    DELETE_PROCEDURE_OR_SERVICE = 'PLAN_DELETE_PROCEDURE_OR_SERVICE',

    ADD_REFERRAL = 'PLAN_ADD_REFERRAL',
    UPDATE_REFERRAL_DEPARTMENT = 'PLAN_UPDATE_REFERRAL_DEPARTMENT',
    UPDATE_REFERRAL_COMMENTS = 'PLAN_UPDATE_REFERRAL_COMMENTS',
    UPDATE_REFERRAL_WHEN = 'PLAN_UPDATE_REFERRAL_WHEN',
    DELETE_REFERRAL = 'PLAN_DELETE_REFERRAL',

    UPDATE_ADMIT_TO_HOSPITAL = 'PLAN_UPDATE_ADMIT_TO_HOSPITAL',
    UPDATE_EMERGENCY = 'PLAN_UPDATE_EMERGENCY',
    UPDATE_SICKNESS_LEVEL = 'PLAN_UPDATE_SICKNESS_LEVEL',
}

export enum HPI_ACTION {
    PROCESS_KNOWLEDGE_GRAPH = 'HPI_PROCESS_KNOWLEDGE_GRAPH',
    BODY_LOCATION_RESPONSE = 'BODY_LOCATION_RESPONSE',
    BODY_LOCATION_HANDLE_TOGGLE = 'HPI_BODY_LOCATION_HANDLE_TOGGLE',
    SINGLE_MULTIPLE_CHOICE_HANDLE_CLICK = 'SINGLE_MULTIPLE_CHOICE_HANDLE_CLICK',
    HANDLE_INPUT_CHANGE = 'HPI_HANDLE_INPUT_CHANGE',
    HANDLE_NUMERIC_INPUT_CHANGE = 'HANDLE_NUMERIC_INPUT_CHANGE',
    LIST_TEXT_HANDLE_CHANGE = 'HPI_LIST_TEXT_HANDLE_CHANGE',
    REMOVE_LIST_INPUT = 'HPI_LIST_TEXT_REMOVE_INPUT',
    ADD_LIST_INPUT = 'HPI_LIST_TEXT_ADD_INPUT',
    HANDLE_TIME_INPUT_CHANGE = 'HPI_HANDLE_TIME_INPUT_CHANGE',
    HANDLE_TIME_OPTION_CHANGE = 'HPI_HANDLE_TIME_OPTION_CHANGE',
    HANDLE_TIME_OF_DAY_CHANGE = 'HPI_HANDLE_TIME_OF_DAY_CHANGE',
    YES_NO_TOGGLE_OPTION = 'HPI_YES_NO_TOGGLE_OPTION',
    SCALE_HANDLE_VALUE = 'HPI_SCALE_HANDLE_VALUE',
    SCALE_HANDLE_CLEAR = 'HPI_SCALE_HANDLE_CLEAR',
    HANDLE_BLANK_QUESTION_CHANGE = 'HPI_HANDLE_BLANK_QUESTION_CHANGE',
    POP_RESPONSE = 'HPI_POP_RESPONSE',
    LAB_TEST_INPUT_CHANGE = 'HPI_LAB_TEST_INPUT_CHANGE',
    LAB_TEST_HANDLE_CLICK = 'HPI_LAB_TEST_HANDLE_CLICK',
    MEDS_POP_YES_NO_TOGGLE = 'HPI_MEDS_POP_YES_NO_TOGGLE',
}

export enum CHIEF_COMPLAINTS {
    SELECT_CHIEF_COMPLAINTS = 'SELECT_CHIEF_COMPLAINTS',
    SET_NOTES_CHIEF_COMPLAINTS = 'SET_NOTES_CHIEF_COMPLAINTS',
}

export enum PATIENT_INFORMATION_ACTION {
    UPDATE_INFORMATION = 'UPDATE_INFORMATION',
    UPDATE_PRONOUNS = 'UPDATE_PRONOUNS',
    UPDATE_NAME = 'UPDATE_NAME',
}

export enum HPI_HEADER_ACTION {
    SAVE_HPI_HEADER = 'SAVE_HPI_HEADER',
}
