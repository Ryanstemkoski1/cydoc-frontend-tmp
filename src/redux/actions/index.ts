import { ROSActionTypes } from './reviewOfSystemsActions';
import { MedicalHistoryActionTypes } from './medicalHistoryActions';
import { SurgicalHistoryActionTypes } from './surgicalHistoryActions';
import { MedicationsActionTypes } from './medicationsActions';
import { AllergiesActionTypes } from './allergiesActions';
import { SocialHistoryActionTypes } from './socialHistoryActions';
import { FamilyHistoryActionTypes } from './familyHistoryActions';
import { PhysicalExamActionTypes } from './physicalExamActions';
import { PlanActionTypes } from './planActions';
import { HpiActionTypes } from './hpiActions';
import { WidgetActionTypes } from './widgetActions';
import { chiefComplaintsActionTypes } from './chiefComplaintsActions';

export type AllActionTypes =
    | ROSActionTypes
    | PhysicalExamActionTypes
    | WidgetActionTypes
    | MedicalHistoryActionTypes
    | SurgicalHistoryActionTypes
    | MedicationsActionTypes
    | AllergiesActionTypes
    | SocialHistoryActionTypes
    | FamilyHistoryActionTypes
    | PlanActionTypes
    | HpiActionTypes
    | chiefComplaintsActionTypes;
