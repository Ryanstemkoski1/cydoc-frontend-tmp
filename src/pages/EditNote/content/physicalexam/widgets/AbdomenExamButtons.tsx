import React, { Component } from 'react';
import { Button, ButtonProps } from 'semantic-ui-react';
import { connect, ConnectedProps } from 'react-redux';
import '../PhysicalExam.css';
import { CurrentNoteState } from 'redux/reducers';
import {
    AbdomenWidgetSection,
    AbdomenWidgetState,
} from 'redux/reducers/widgetReducers/abdomenWidgetReducer';
import { toggleAbdomenWidgetSection } from 'redux/actions/widgetActions/abdomenWidgetActions';
import { selectAbdomenWidgetSection } from 'redux/selectors/widgetSelectors/abdomenWidgetSelectors';

const options: (keyof AbdomenWidgetSection)[] = [
    'tenderness',
    'rebound',
    'guarding',
];

interface AbdomenExamButtonsProps {
    abdomenQuadrant: keyof AbdomenWidgetState;
}

class AbdomenExamButtons extends Component<
    AbdomenExamButtonsProps & PropsFromRedux
> {
    getHandleClick = (fieldName: keyof AbdomenWidgetSection) => (
        event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
        _data: ButtonProps
    ) => {
        event.preventDefault();
        this.props.toggleSection(this.props.abdomenQuadrant, fieldName);
    };

    renderButton = (option: keyof AbdomenWidgetSection) => (
        <Button
            size='small'
            key={option}
            active={this.props.sections[option]}
            color={this.props.sections[option] ? 'red' : undefined}
            content={option}
            onClick={this.getHandleClick(option)}
            className='pe-ros-button spaced-buttons'
        />
    );

    render() {
        return <div>{options.map(this.renderButton)}</div>;
    }
}

const mapStateToProps = (
    state: CurrentNoteState,
    ownProps: AbdomenExamButtonsProps
) => ({
    sections: selectAbdomenWidgetSection(state, ownProps.abdomenQuadrant),
});

const mapDispatchToProps = {
    toggleSection: toggleAbdomenWidgetSection,
};

const connector = connect(mapStateToProps, mapDispatchToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;

export default connector(AbdomenExamButtons);
