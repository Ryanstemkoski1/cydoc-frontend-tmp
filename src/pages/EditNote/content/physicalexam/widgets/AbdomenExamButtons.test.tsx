import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import AbdomenExamButtons from './AbdomenExamButtons';
import { initialPhysicalExamState } from 'redux/reducers/physicalExamReducer';
import { AbdomenWidgetState } from 'redux/reducers/widgetReducers/abdomenWidgetReducer';
import { ABDOMEN_WIDGET_ACTION } from 'redux/actions/actionTypes';
import { currentNoteStore } from 'redux/store';

const section = 'rightUpperQuadrant';
const field = 'tenderness';
const mockStore = configureStore([]);

const mountWithMockStore = (
    initStore = {
        physicalExam: initialPhysicalExamState,
    },
    abdomenQuadrant: keyof AbdomenWidgetState = section,
    { ...props } = {}
) => {
    const store = mockStore(initStore);
    return {
        store,
        wrapper: mount(
            <Provider store={store}>
                <AbdomenExamButtons
                    abdomenQuadrant={abdomenQuadrant}
                    {...props}
                />
            </Provider>
        ),
    };
};

const mountWithRealStore = (
    abdomenQuadrant: keyof AbdomenWidgetState = section,
    { ...props } = {}
) => {
    return mount(
        <Provider store={currentNoteStore}>
            <AbdomenExamButtons abdomenQuadrant={abdomenQuadrant} {...props} />
        </Provider>
    );
};

describe('AbdomenExamButtons', () => {
    let wrapper: ReactWrapper;

    beforeEach(() => {
        ({ wrapper } = mountWithMockStore());
    });

    it('renders without crashing', () => {
        expect(wrapper).toBeTruthy();
    });

    it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});

describe('AbdomenExamButtons with mock store', () => {
    it('dispatches correct action when clicking an option', () => {
        const { store, wrapper } = mountWithMockStore();
        wrapper.findWhere((node) => node.key() === field).simulate('click');

        const expectedActions = [
            {
                type: ABDOMEN_WIDGET_ACTION.TOGGLE_ABDOMEN_WIDGET_SECTION,
                payload: {
                    section,
                    field,
                },
            },
        ];
        expect(store.getActions()).toEqual(expectedActions);
    });
});

describe('AbdomenExamButtons with real store', () => {
    let wrapper: ReactWrapper;
    const findButton = (key: string) =>
        wrapper.findWhere((node) => node.key() === key);

    beforeEach(() => {
        wrapper = mountWithRealStore();
    });

    it('changes button color after being clicked', () => {
        expect(findButton(field).prop('color')).toBe(undefined);

        findButton(field).simulate('click');
        expect(findButton(field).prop('color')).toBe('red');

        findButton(field).simulate('click');
        expect(findButton(field).prop('color')).toBe(undefined);
    });
});
