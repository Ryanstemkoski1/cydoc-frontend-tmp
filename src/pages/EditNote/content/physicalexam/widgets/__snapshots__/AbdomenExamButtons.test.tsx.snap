// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbdomenExamButtons matches snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(AbdomenExamButtons)
    abdomenQuadrant="rightUpperQuadrant"
  >
    <AbdomenExamButtons
      abdomenQuadrant="rightUpperQuadrant"
      sections={
        Object {
          "guarding": false,
          "rebound": false,
          "tenderness": false,
        }
      }
      toggleSection={[Function]}
    >
      <div>
        <Button
          as="button"
          className="spaced-buttons"
          content="tenderness"
          key="tenderness"
          onClick={[Function]}
          size="small"
        >
          <Ref
            innerRef={
              Object {
                "current": <button
                  class="ui small button spaced-buttons"
                >
                  tenderness
                </button>,
              }
            }
          >
            <RefFindNode
              innerRef={
                Object {
                  "current": <button
                    class="ui small button spaced-buttons"
                  >
                    tenderness
                  </button>,
                }
              }
            >
              <button
                className="ui small button spaced-buttons"
                onClick={[Function]}
              >
                tenderness
              </button>
            </RefFindNode>
          </Ref>
        </Button>
        <Button
          as="button"
          className="spaced-buttons"
          content="rebound"
          key="rebound"
          onClick={[Function]}
          size="small"
        >
          <Ref
            innerRef={
              Object {
                "current": <button
                  class="ui small button spaced-buttons"
                >
                  rebound
                </button>,
              }
            }
          >
            <RefFindNode
              innerRef={
                Object {
                  "current": <button
                    class="ui small button spaced-buttons"
                  >
                    rebound
                  </button>,
                }
              }
            >
              <button
                className="ui small button spaced-buttons"
                onClick={[Function]}
              >
                rebound
              </button>
            </RefFindNode>
          </Ref>
        </Button>
        <Button
          as="button"
          className="spaced-buttons"
          content="guarding"
          key="guarding"
          onClick={[Function]}
          size="small"
        >
          <Ref
            innerRef={
              Object {
                "current": <button
                  class="ui small button spaced-buttons"
                >
                  guarding
                </button>,
              }
            }
          >
            <RefFindNode
              innerRef={
                Object {
                  "current": <button
                    class="ui small button spaced-buttons"
                  >
                    guarding
                  </button>,
                }
              }
            >
              <button
                className="ui small button spaced-buttons"
                onClick={[Function]}
              >
                guarding
              </button>
            </RefFindNode>
          </Ref>
        </Button>
      </div>
    </AbdomenExamButtons>
  </Connect(AbdomenExamButtons)>
</Provider>
`;
